File: Week_08/Practicum/Find Occurrences of an Element in an Array.cpp
class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
        vector<int> result(queries.size(), -1);
        unordered_map<int, vector<int>> positions;
        for (int i = 0; i < nums.size(); i++){
            positions[nums[i]].push_back(i);
        }

        if (positions.count(x) == 0) 
            return result;

        for (int i = 0; i < queries.size(); i++){
            if (queries[i] <= positions[x].size()) {
                result[i] = positions[x][queries[i] - 1];
            }
        }

        return result;
    }
};

class Solution {
public:
    vector<int> occurrencesOfElement(vector<int>& nums, vector<int>& queries, int x) {
        std::vector<int> occurrences;
        for (int i = 0; i < nums.size(); ++i) {
            if (nums[i] == x) {
                occurrences.push_back(i);
            }
        }

        std::vector<int> answers(queries.size(), -1);
        for (int i = 0; i < queries.size(); ++i) {
            if (queries[i] <= occurrences.size()) {
                answers[i] = occurrences[queries[i] - 1];
            }
        }

        return answers;
    }
};

File: Week_08/Practicum/Find Occurrences of an Element in an Array.py
class Solution:
    def occurrencesOfElement(self, nums: List[int], queries: List[int], x: int) -> List[int]:
        d = dict()
        counter = 1
        for i, num in enumerate(nums):
            if num == x:
                d[counter] = i
                counter += 1

        ans = []

        for query in queries:
            if query in d.keys():
                ans.append(d[query])
            else:
                ans.append(-1)

        return ans